lib/node_modules/@stdlib/stats/array/range-by/test/test.js

```diff
@@ -0,0 +1,289 @@
+
+	t.end();
+});
+
+tape( 'the function calculates the range of an array via a callback function (accessors)', function test( t ) {
+	var x;
+	var v;
+
+	x = [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ];
+	v = rangeBy( toAccessorArray( x ), accessor );
+	t.strictEqual( v, 18.0, 'returns expected value' );
+
+	x = [ -4.0, -5.0 ];
+	v = rangeBy( toAccessorArray( x ), accessor );
+	t.strictEqual( v, 2.0, 'returns expected value' );
+
+	x = [ -0.0, 0.0, -0.0 ];
+	v = rangeBy( toAccessorArray( x ), accessor );
+	t.strictEqual( isPositiveZero( v ), true, 'returns expected value' );
+
+	x = [ NaN ];
+	v = rangeBy( toAccessorArray( x ), accessor );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	x = [ NaN, NaN ];
+	v = rangeBy( toAccessorArray( x ), accessor );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function calculates the range of an array (array-like object)', function test( t ) {
+	var x;
+	var v;
+
+	x = {
+		'length': 6,
+		'0': 1.0,
+		'1': -2.0,
+		'2': -4.0,
+		'3': 5.0,
+		'4': 0.0,
+		'5': 3.0
+	};
+	v = rangeBy( x, accessor );
+	t.strictEqual( v, 18.0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function supports providing a callback execution context', function test( t ) {
+	var expected;
+	var indices;
+	var values;
+	var arrays;
+	var ctx;
+	var x;
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	ctx = {
+		'count': 0
+	};
+	indices = [];
+	values = [];
+	arrays = [];
+	rangeBy( x, accessor, ctx );
+
+	t.strictEqual( ctx.count, x.length, 'returns expected value' );
+
+	expected = [ 0, 1, 2, 3, 4 ];
+	t.deepEqual( indices, expected, 'returns expected value' );
+
+	expected = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	t.deepEqual( values, expected, 'returns expected value' );
+
+	expected = [ x, x, x, x, x ];
+	t.deepEqual( arrays, expected, 'returns expected value' );
+	t.end();
+
+	function accessor( v, idx, arr ) {
+		this.count += 1; // eslint-disable-line no-invalid-this
+		indices.push( idx );
+		values.push( v );
+		arrays.push( arr );
+		return v * 2.0;
+	}
+});
+
+tape( 'the function supports providing a callback execution context (accessors)', function test( t ) {
+	var expected;
+	var indices;
+	var values;
+	var arrays;
+	var ctx;
+	var ax;
+	var x;
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	ax = toAccessorArray( x );
+	ctx = {
+		'count': 0
+	};
+	indices = [];
+	values = [];
+	arrays = [];
+	rangeBy( ax, accessor, ctx );
+
+	t.strictEqual( ctx.count, x.length, 'returns expected value' );
+
+	expected = [ 0, 1, 2, 3, 4 ];
+	t.deepEqual( indices, expected, 'returns expected value' );
+
+	expected = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	t.deepEqual( values, expected, 'returns expected value' );
+
+	expected = [ ax, ax, ax, ax, ax ];
+	t.deepEqual( arrays, expected, 'returns expected value' );
+	t.end();
+
+	function accessor( v, idx, arr ) {
+		this.count += 1; // eslint-disable-line no-invalid-this
+		indices.push( idx );
+		values.push( v );
+		arrays.push( arr.data );
```

Comment:
@gururaj1512 This isn't correct. Something is off here. The callback should be provided the input array. If this isn't true, then the upstream implementation is incorrect.